pkg/fields/selector.go

```diff
@@ -75,6 +82,14 @@ func (t *hasTerm) Transform(fn TransformFunc) (Selector, error) {
 	return &hasTerm{field, value}, nil
 }
 
+func (t *hasTerm) Requirements() []labels.Requirement {
+	r, err := labels.NewRequirement(t.field, labels.EqualsOperator, sets.NewString(t.value))
+	if err != nil {
+		panic(fmt.Sprintf("Impossible to create invalid requirement label. Err: %v", err))
```

Comment:
NewRequirement validates input. In this case, EqualsOperator only allows `len(values) = 1`, which we explicitly ensure.

I think this is not valid either. Field syntax is different from label syntax. We should have another method to convert field selector into Requirements. Will change it shortly.